### PR TITLE
Automation: GUI infra phase 1

### PR DESCRIPTION
**Pull request by psiinon**
Some initial changes to the AF infrastructure required for the GUI - there will be more...
I'm working on the GUI in parallel but wont submit that until its in a more suitable state ;)
The main changes here are:
* Process the test data in the test constructor
* Introduce test runTest() and hasRun() methods distinct from hasPassed()
* Introduce an AutomationPlan class to tie everything together
* Associated job data with jobs

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by psiinon** [comment]
For info the following add-ons will need to be released together: automation, graphql, openapi, reports, soap, spiderajax, wappalyzer

I've hopefully updated them all correctly, ie to depend on the right automation version. I've put it down as "Maintenance changes" but can add a specific comment if people think thats better...

**Comment by psiinon** [comment]
I've added the changes to associate the job data with the jobs and no more changes planned, so hopefully its ready to review :)

**Comment by psiinon** [comment]
FYI my GUI code accesses the container contents that LGTM is complaining about, so I was going to ignore that alert ;)

**Comment by thc202** [comment]
Has merge conflicts now.

**Comment by psiinon** [comment]
Conflict fixed but I may need to make another change based on more GUI requirements...

**Comment by psiinon** [comment]
Changed the abstract test to maintain the state as thats something that all tests will need to do and added the reset() method.
Should be ready for review again...

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java [hunk 1]
```diff
@@ -72,10 +73,27 @@ public void setName(String name) {
...
+    /*
+        public void verifyJobSpecificData(LinkedHashMap<?, ?> jobData, AutomationProgress progress) {
+        	this.verifyJobSpecificData(progress);
+        }
+
+        public void runJob(
+                AutomationEnvironment env, LinkedHashMap<?, ?> jobData, AutomationProgress progress) {
+        	this.runJob(env, progress);
+        }
+    */
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java:93]: Remove?

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationPlan.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        this.progress = progress;
+        this.env = env;
+        this.jobs = jobsToRun;
+    }
+
+    public AutomationProgress getProgress() {
+        return progress;
+    }
+
+    public AutomationProgress resetProgress() {
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationPlan.java:42]: Does it really need to return the progress? There's already a getter.

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/ExtensionAutomation.java [hunk 3]
```diff
@@ -219,46 +261,33 @@ private AutomationProgress runPlan(LinkedHashMap<?, ?> envData, ArrayList<?> job
                     continue;
                 }
                 job.setEnv(env);
-                job.verifyParameters((LinkedHashMap<?, ?>) paramsObj, progress);
-                job.verifyJobSpecificData(jobData, progress);
-                jobsToRun.put(job, jobData);
+                job.setJobData(jobData);
+                job.verifyParameters(progress);
+                job.verifyJobSpecificData(progress);
+                // jobsToRun.put(job, jobData);
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/ExtensionAutomation.java:267]: Remove?

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/RequestorJob.java [hunk 4]
```diff
@@ -62,7 +62,11 @@ public RequestorJob(HttpSender httpSender) {
     }
 
     @Override
-    public void verifyJobSpecificData(LinkedHashMap<?, ?> jobData, AutomationProgress progress) {
+    public void verifyJobSpecificData(AutomationProgress progress) {
+        LinkedHashMap<?, ?> jobData = this.getJobData();
+        if (jobData == null) {
+            return;
+        }
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/RequestorJob.java:69]: Should the other jobs do the same?

File: addOns/automation/src/test/java/org/zaproxy/addon/automation/AutomationProgressUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+class AutomationProgressUnitTest extends TestUtils {
+
+    private AutomationProgress progress;
+
+    @BeforeEach
+    void setUp() {
+        progress = new AutomationProgress();
+    }
+
+    static Stream<Arguments> shouldPassTestParamsSource() {
```
Comment (thc202) [inline addOns/automation/src/test/java/org/zaproxy/addon/automation/AutomationProgressUnitTest.java:42]: Remove?

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/RequestorJob.java (hunk 4 above)
Comment (psiinon) [reply addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/RequestorJob.java:69]: Depends if they require it ;) I'll check them all...

Comment (psiinon) [reply addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/RequestorJob.java:69]: Ah no, it cant be null :)

**Comment by psiinon** [comment]
Hopefully all points addressed :)

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]
Thanks

**Pull request merged**